cider/exec/template/operator/join/hashtable/BaselineJoinHashTable.cpp

```diff
@@ -284,7 +284,7 @@ std::pair<size_t, size_t> BaselineJoinHashTable::approximateTupleCount(
               << ", emitted keys count: " << cached_count_info.second;
       return std::make_pair(*cached_count_info.first, cached_count_info.second);
     }
-    int thread_count = cpu_threads();
+    int thread_count = 1;
```

Comment:
Hmm, can we make this a configurable option (e.g., a config in https://github.com/intel/BDTK/blob/eda3ea46087b9ba1893f5d8e520814ca4281fca4/cider/include/cider/CiderOptions.h#L105)instead of hardcode?

Reply:
I think it's not necessary.

Reply:
As a database, multi-thread makes senses. But for cider, I thought it just should be one thread.

Reply:
It's common to have multi-threading hash table builder. But current usage of Cider is quite limited (requiring all data ready).  It's a boarder discussion. Let's have a follow-up discussion with @chaojun-zhang.

Back to this PR. LGTM to merge this first.